### DISCUSSION TITLE
Invisible data in the root of a pool?

### DISCUSSION THREAD

**Discussion by closewall** [openzfs/zfs#15070] [Q&A] [answered]
Hello.
I have a problem where there appears to be invisible data at the root of my pool. When I run `zfs list`, I see:

```
~# zfs list

NAME                   USED  AVAIL     REFER  MOUNTPOINT
ironbox               1.16T  30.4T      161G  /mnt/ironbox
ironbox/zfs_Backup    1022G  30.4T     1022G  /mnt/ironbox/zfs_Backup
ironbox/zfs_Photos     140K  30.4T      140K  /mnt/ironbox/zfs_Photos
```

Notice how 161G is referenced under ironbox (which is the name of my pool).

And if I list the files:

```
/mnt/ironbox# ls
zfs_Backup/  zfs_Photos/
```

I only see my two datasets.

How do I get rid of this 161G of invisible data? Is my only option to re-create the entire pool?

For what it's worth, I am certain this is data I tried to rsync from a zfs filesystem on different hardware. (I couldn't get zfs send working so decided to use rsync).
I am positive I specified the output directory to the "/zfs_Photos" dataset in my rsync command, and it worked normally. But for some reason, that dataset is empty, and all that data seems to be occupying space in the root of my pool.

I did not create any snapshots. This is a brand new pool.

**Comment by rincebrain** [comment, marked as answer]
If the dataset wasn't mounted when you did the rsync, it would have written it to the folder "/zfs_Photos" in the dataset "ironbox", not to the folder "/" in the dataset "ironbox/zfs_Photos".

So unmount ironbox/zfs_Photos, and I would bet your missing gigabytes will appear.

**Comment by closewall** [comment]
> If the dataset wasn't mounted when you did the rsync, it would have written it to the folder "/zfs_Photos" in the dataset "ironbox", not to the folder "/" in the dataset "ironbox/zfs_Photos".
> 
> So unmount ironbox/zfs_Photos, and I would bet your missing gigabytes will appear.

Thanks, unmounting and mounting worked (eventually).

**Comment by slavanap** [comment]
Snapshots are not shown with that command. It order to understand what does take size more precisely, try `zfs list -rtall`

**Reply by rincebrain** [reply]
Since the size differential is included in REFER of the root dataset, snapshots are not involved, that is data referenced in the current version of the dataset.

**Reply by slavanap** [reply]
@rincebrain you're right. My point is on USED sum up snapshots that you won't be able to see with `zfs list` command. As for "hidden" data, it's there either in a form of dotfiles/folders, or covered by another mountpoints (as stated in another answer above) or in a form of deleted files which are still being opened (`sudo lsof | grep your_folder_path` may help to identify those)

**Reply by rincebrain** [reply]
They already replied saying I was right about that above.
